src/main/java/subway/section/SectionDao.java

```diff
@@ -0,0 +1,57 @@
+package subway.section;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import subway.line.Line;
+import subway.line.LineDao;
+import subway.station.StationDao;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+
+import static subway.section.SectionQuery.*;
+
+@Repository
+public class SectionDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    public SectionDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public Section save(Section section) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(con -> {
+            PreparedStatement psmt = con.prepareStatement(
+                    insertQuery,
+                    Statement.RETURN_GENERATED_KEYS);
+            psmt.setLong(1, section.getLine().getId());
+            psmt.setLong(2, section.getUpStation().getId());
+            psmt.setLong(3, section.getDownStation().getId());
+            psmt.setInt(4, section.getDistance());
+            return psmt;
+        }, keyHolder);
+
+        Long id = (Long) keyHolder.getKey();
+
+        return new Section(
+                id,
+                section.getLine(),
+                section.getUpStation(),
+                section.getDownStation(),
+                section.getDistance()
+        );
+    }
+
+    public SectionsInLine findAllByLine(Line line, StationDao stationDao, LineDao lineDao) {
+        return new SectionsInLine(jdbcTemplate.query(selectByIdQuery, new SectionMapper(stationDao, lineDao), line.getId()));
```

Comment:
질문에 적어 주신 것처럼 이렇게 의존성이 생겨 버렸군요!
일단 Section 이라는 도메인 자체가 일종의 mapping 관계를 가지는 도메인이라 이것도 나쁘지는 않다고 생각은 하는데요 
JPA 에서 정의된  eager fetch 랑 비슷하기도 하구요 
물론 JPA 에서 dao 객체에서 다른 dao 를 참조하는 방식은 아니고 일단 new Station(rs.getLong("up_station_id")) 
만 해서 리턴한 후에 나중에 station 을 조회 해서 station 의 내부 데이터를 다시 채우는 방식입니다.

위처럼 다시 수정을 해봐도 좋을것 같구요 저는 지금도 나쁘지는 않다고 생각해요 
다만 Mapper 까지 dao 의존성이 넘어가고 있고 Line 같은 경우에는 다시 조회 할 필요가 없을것 같아서
저는 이렇게 수정을 할것 같아요 

```suggestion
        return new SectionsInLine(jdbcTemplate.query(selectByIdQuery, (rs, rowNum) -> {
            Long id = rs.getLong("id");
            Station upStation = stationDao.findById(rs.getLong("up_station_id")).orElseThrow(NotFoundException::new);
            Station downStation = stationDao.findById(rs.getLong("down_station_id")).orElseThrow(NotFoundException::new);
            int distance = rs.getInt("distance");

            return new Section(id, line, upStation, downStation, distance);
        }));
```

지금만들어 놓은 SectionMapper 를 다른곳에서 사용하게 될때 상황마다 구현이 달라지는 것도 좀 걸리고 
딱 이상황에서만 사용될 로직이라고 판단이 되고 line 도 바로 세팅 할구 있도록 람다 표현식을 사용하면 조금은 괜찮을것 같습니다.
이건 제 의견이고요 
jon 이 생각하는 것으로 결정하시면 될 것 같습니다.